pkg/certificate/controller.go

```diff
@@ -87,107 +102,31 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (m *Manager) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := m.log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name).WithName("Reconcile")
-	reqLogger.Info("Reconciling Certificates")
-
-	elapsedToRotateCA := m.elapsedToRotateCAFromLastDeadline()
-	elapsedToRotateServices := m.elapsedToRotateServicesFromLastDeadline()
-
-	// Ensure that this Reconcile is not called after bad changes at
-	// the certificate chain
-	if elapsedToRotateCA > 0 {
-		err := m.verifyTLS()
-		if err != nil {
-			reqLogger.Info(fmt.Sprintf("TLS certificate chain failed verification, forcing rotation, err: %v", err))
-			// Force rotation
-			elapsedToRotateCA = 0
-		}
-	}
-
-	// We have pass expiration time for the CA
-	if elapsedToRotateCA <= 0 {
+	reqLogger.Info("Reconciling Certificates after request")
 
-		// If rotate fails runtime-controller manager will re-enqueue it, so
-		// it will be retried
-		err := m.rotateAll()
-		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed rotating all certs")
-		}
-
-		// Re-calculate elapsedToRotate since we have generated new
-		// certificates
-		m.nextRotationDeadlineForCA()
-		elapsedToRotateCA = m.elapsedToRotateCAFromLastDeadline()
-
-		// Also recalculate it for serices certificate since they has changed
-		m.nextRotationDeadlineForServices()
-		elapsedToRotateServices = m.elapsedToRotateServicesFromLastDeadline()
-
-	} else if elapsedToRotateServices <= 0 {
-		// CA is ok but expiration but we have passed expiration time for service certificates
-		err := m.rotateServicesWithOverlap()
-		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed rotating services certs")
-		}
-
-		// Re-calculate elapsedToRotateServices since we have generated new
-		// services certificates
-		m.nextRotationDeadlineForServices()
-		elapsedToRotateServices = m.elapsedToRotateServicesFromLastDeadline()
-	}
+	objects := objectMap{}
+	certificateChain := chain.CertificateChain{}
 
-	elapsedForCABundleCleanup, err := m.earliestElapsedForCACertsCleanup()
+	err := m.readCertificateChain(objects, &certificateChain)
```

Comment:
What do you think about moving al the Code from  `Reconcile` to `Manager.reconcileCertificates` this way we cleanup a little the already bloated controller (a lot of this code is boilerplate)

Apart from that why don't we make `readCertificateChain` return directly objects and chain instead of needed them as arguments

Reply:
> What do you think about moving al the Code from `Reconcile` to `Manager.reconcileCertificates` this way we cleanup a little the already bloated controller (a lot of this code is boilerplate)

OK

> Apart from that why don't we make `readCertificateChain` return directly objects and chain instead of needed them as arguments

It is more clear to me the usage of this kind of interface that the intention of these methods (`read/update/write`) is to make modifications in place, even though for some might not apply. The commonality of it all makes it clear that everything happens in place of those to arguments throughout all operations.